server/handlers/handlers.go

```diff
@@ -349,6 +350,14 @@ func (ah *ActivityHandler) handleCreatedActivity(msg *clientmodels.Message, subs
 		senderID = ah.plugin.GetBotUserID()
 	}
 
+	if ah.plugin.GetSyncRemoteOnly() {
```

Comment:
Should this be "inside" the check for `plugin.SelectiveSync()` so that it only applies when the latter is enabled?